apps/storybook.namekit.io/stories/Namekit/Identity.stories.tsx

```diff
@@ -35,29 +35,147 @@ const IdentityCard: React.FC<{
     <Identity.Name />
     <Identity.Address />
     <Identity.NameGuardShield />
-    <Identity.ENSProfileLink />
+    <Identity.ProfileLink>
+      <div>
+        <ENSLogo />
+        <span>View on ENS App</span>
+      </div>
+    </Identity.ProfileLink>
     <Identity.Followers />
   </Identity.Root>
 );
 
+const customAppConfig = {
+  profileLinks: {
+    getProfileURL: (address: string) => `/profiles/${address}`,
+    getProfileLink: (address: string, children: React.ReactNode) => (
+      <a href={`/profiles/${address}`}>{children}</a>
+    ),
+  },
+};
+
+const CustomAppIdentityCard: React.FC<{ address: string }> = ({ address }) => (
+  <NameKitProvider config={customAppConfig}>
+    <Identity.Root address={address}>
```

Comment:
Suggest we make a few updates here to put our "Identity" components on a stronger foundation. Specifically: an "identity reference" is more than just an address. It is a combination of a chain and an address.

1. Suggest defining an interface for `ImplicitIdentityReference` that combines both an address and a chain, but where the chain param is optional. The purpose of this interface is to optimize DX for the average case where all addresses within an app are on the same chain. It should be noted how this isn't always the case though, and increasingly won't be in the future the way things are headed.
2. Suggest defining an interface for `IdentityReference` that extends `ImplicitIdentityReference` but makes the chain param required.
3. Suggest defining a little utility function that can be used within the context of a NameKitProvider that automatically converts any `ImplicitIdentityReference` to an explicit `IdentityReference` using the configured chain as the default.
4. There's a bunch of functions here that currently only take an "address" param. These should change so that they instead take a param of type `IdentityReference` or `ImplicitIdentityReference`. Inside the implementation of these functions, we can call the utility function described above that takes either an `IdentityReference` or `ImplicitIdentityReference` as a param and always returns an explicit `IdentityReference` for use in all our internal logic.
5. Functions such as `getProfileURL` and `getProfileLink` should take an `IdentityReference` as a param. The default implementation of these functions should then consider the chain parameter and not only the address. For example. If the chain param is mainnet, then our existing strategy is good. If the chain param is anything else (such as sepolia, etc..), then there isn't an existing profile service we can easily link to for this. So instead suggest making our default implementation of all the profile link generator functions trigger some alert box or something like that for the case that the chain is anything but mainnet. Once we build the backend APIs into NameKit for generating profile data, we can come back to this and implement some improved default behaviour for showing profiles on chains other than mainnet.

Here's an acceptance criteria for everything described above: If I use our Identity components to show the identity for a chain other than mainnet (such as sepolia) I shouldn't get a profile link to view the profile of that address on mainnet as that wouldn't actually be the correct profile / identity.

Please ask me if any questions :+1: